src/platform_impl/macos/app_state.rs

```diff
@@ -180,13 +177,19 @@ impl Handler {
 pub enum AppState {}
 
 impl AppState {
-    pub fn set_callback<F, T>(callback: F, window_target: RootWindowTarget<T>)
+    pub unsafe fn set_callback<F, T>(callback: F, window_target: Rc<RootWindowTarget<T>>)
     where
-        F: 'static + FnMut(Event<T>, &RootWindowTarget<T>, &mut ControlFlow),
-        T: 'static,
+        F: FnMut(Event<T>, &RootWindowTarget<T>, &mut ControlFlow),
     {
         *HANDLER.callback.lock().unwrap() = Some(Box::new(EventLoopHandler {
-            callback,
+            // This transmute is always safe, in case it was reached through `run`, since our
+            // lifetime will be already 'static. In other cases caller should ensure that all data
+            // he passed to callback will actually outlive it, some apps just can't move everything
```

Comment:
```suggestion
            // they passed to callback will actually outlive it, some apps just can't move everything
```
Do girls not use winit?

Reply:
It kind of "thoughts" translation mistake from Russian to English, since your suggestion doesn't maps to Russian (or I find it odd). Will change this one in a moment, sorry.